Assets/Shopify/UIToolkit/Components/WebImageCache.cs

```diff
@@ -0,0 +1,163 @@
+ namespace Shopify.UIToolkit {
+    using UnityEngine;
+    using System;
+    using System.Collections;
+    using System.Collections.Generic;
+
+    public struct CachedWebResource<T> {
+        public readonly string LastModifiedTimestamp;
+        public readonly T Data;
+
+        public CachedWebResource(string lastModifiedTimestamp, T data) {
+            LastModifiedTimestamp = lastModifiedTimestamp;
+            Data = data;
+        }
+    }
+
+    /// <summary>
+    /// This class implements a LRU cache for images (PNG/JPEG) that were downloaded from the web.
+    /// </summary>
+    public class WebImageCache {
+        /// <summary>
+        /// The default estimated memory size limit in bytes.
+        /// </summary>
+        public const int DEFAULT_MEMORY_SIZE_LIMIT = 20971520; // 20 MB in bytes
+
+        private Dictionary<string, CachedWebResource<Texture2D>> _urlTextureCache = new Dictionary<string, CachedWebResource<Texture2D>>();
+        private LinkedList<string> _recentlyUsed = new LinkedList<string>();
+        private int _memorySizeLimit;
+        private int _currentMemoryFootprint;
+
+        private static WebImageCache _sharedCache;
+
+        /// <summary>
+        /// Shared WebImageCache instanced.
+        /// </summary>
+        /// <returns>A singleton instance of the WebImageCache.</returns>
+        public static WebImageCache SharedCache {
+            get {
+                _sharedCache = _sharedCache != null ? _sharedCache : new WebImageCache(DEFAULT_MEMORY_SIZE_LIMIT);
```

Comment:
you can use null propogation here `_sharedCache = _sharedCache ?? new Whatever()`, its a lil more concise?